torch_xla/csrc/runtime/cache.h

```diff
@@ -106,6 +122,116 @@ class Cache {
   ElementMap element_map_;
 };
 
+// A persistent cache which serializes values to disk. This wraps a Cache
+// instance, so values will only be read from disk once and subsequent reads
+// will go through the wrapped Cache.
+template <typename K, typename T, typename H = std::hash<K>,
+          typename E = std::equal_to<K>>
+class PersistentCache : public AbstractCache<K, T, H, E> {
+ public:
+  using TypePtr = std::shared_ptr<T>;
+
+  explicit PersistentCache(int kMaxMemoryCacheSize, std::string cache_dir,
+                           bool readonly,
+                           std::function<std::string(TypePtr&)> serialize,
```

Comment:
nit. do you expect the serializer to change the original data? otherwise, use `const TypePtr&`